### DISCUSSION TITLE
Could we count group by service(ip) and exception

### DISCUSSION THREAD

**Discussion by yqjdcyy** [apache/skywalking#10706] [Q&A] [answered]
- I try to statistic the count of exception group by service/ip and exception.  the result maybe `service_fail_count{server="server-A", ip="127.0.0.1" exception="java.lang.IllegalArgumentException"} 5 1681956343`
- I found it fail to generate by **OAL** automatically, like `service_fail_count = from(Service.*).filter(status == false).count()` and `service_fail_http_count = from(Service.*).filter(type == RequestType.HTTP).filter(status == false).count()`. Because it need SOURCE and POINTED type.  And **MAL** or **LAL** do not apply to scenarios.
- It seems that **Trace Kafka Exporter** is the last way?

**Comment by wu-sheng** [comment]
I can't follow your case. OAL and MAL are two different things. One for streaming data, like trace and envoy ALS, the other is for metrics.
What you wrote seems to be not clear about where the data is from.

**Reply by yqjdcyy** [reply]
Try to generate metrics from Trace.log, in my option, OAL is the only way.
OAL rules execute with key.getter from @ScopeDeclaration such as egment(segmentId, traceId, serviceId, dataBinary...), but log.data in dataBinary could not read by oal.rule directly

**Reply by wu-sheng** [reply]
I don't think we support that part.

**Reply by wu-sheng** [reply, marked as answer]
OAL is based on the Source(s), rather than trace or something. There is no OAL source associated with `trace.span.logs` AFAIK.

**Reply by yqjdcyy** [reply]
get it, thanks
